lib/screens/login/login_view_model.dart

```diff
@@ -16,4 +18,36 @@ class LoginViewModel extends StateNotifier<void> {
   bool isValidPassword(String? password) {
     return !(password == null || password.length < 8);
   }
+
+  login({required String email, required String password}) async {
+    state = const AsyncLoading();
+    // TODO: Integrate with API
+
+    // Handling error part:
+
+    // If it returns unauthorized error (401)
+    //state = const AsyncError(
+    //  AppError.unauthorized,
+    //  StackTrace.empty,
+    //);
+
+    // If it returns timeout error, then check Internet connection
+    internetConnectionManager = ref.read(internetConnectionManagerProvider);
+    final isConnected = await internetConnectionManager.hasConnection();
+
+    if (!isConnected) {
+      state = const AsyncError(
```

Comment:
May I ask if we need to write unit tests for these to increase the code coverage? as the Codevov just got failed 🙏 
https://github.com/nimblehq/flutter-ic-khanh-thieu/pull/55/checks?check_run_id=15928008166

Reply:
I noticed this too. However, the `login()` method is not integrated with API yet. I think we can supplement unit tests for `LoginViewModel` in the integrating task 🙏